src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,162 @@
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequest(
+            @Parameter(name = "professorName") @RequestParam String professorName,
+            @Parameter(name = "professorEmail") @RequestParam String professorEmail,
+            @Parameter(name = "requesterName") @RequestParam String requesterName,
+            @Parameter(name = "recommendationTypes") @RequestParam String recommendationTypes,
+            @Parameter(name = "details") @RequestParam String details,
+            @Parameter(name = "status") @RequestParam String status,
+            @Parameter(name = "submissionDate", description = "date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("submissionDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime submissionDate,
+            @Parameter(name = "completionDate", description = "date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam(value = "completionDate", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime completionDate)
+            throws JsonProcessingException {
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+        // See: https://www.baeldung.com/spring-date-parameters
+
+        log.info("localDateTime={}", submissionDate);
+        log.info("localDateTime={}", completionDate);
+
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        recommendationRequest.setProfessorName(professorName);
+        recommendationRequest.setProfessorEmail(professorEmail);
+        recommendationRequest.setRequesterName(requesterName);
+        recommendationRequest.setRecommendationTypes(recommendationTypes);
+        recommendationRequest.setDetails(details);
+        recommendationRequest.setStatus(status);
+        recommendationRequest.setSubmissionDate(submissionDate);
+        recommendationRequest.setCompletionDate(completionDate);
+
+        RecommendationRequest savedRecommendationRequest = recommendationRequestRepository.save(recommendationRequest);
+
+        return savedRecommendationRequest;
+    }
+
+    /**
+     * Delete a RecommendationRequest
+     * 
+     * @param id the id of the RecommendationRequest to delete
+     * @return a message indicating that the RecommendationRequest was deleted
+     */
+    @Operation(summary = "Delete a RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("")
+    public Object deleteRecommendationRequest(
+            @Parameter(name = "id") @RequestParam Long id) {
+        RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequestRepository.delete(recommendationRequest);
```

Comment:
You need to add code here to prevent one user from deleting some other users request.

Look, for example, at this code from proj-courses:

https://github.com/ucsb-cs156/proj-courses/blob/4bdc2674508428d03cb2dcd2e9077d75cdaf5a3f/src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java#L111

or this code from proj-gauchoride:

https://github.com/ucsb-cs156/proj-gauchoride/blob/cc4adaa0b4c69a0d9de9860304989c62067e8c2e/src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java#L140